src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,26 @@
 import React from 'react';
+import { Status } from '../../types/Status';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { setQuery, setStatus } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const filter = useAppSelector(state => state.filter);
+
+  const handleStatus = (event: React.ChangeEvent<HTMLSelectElement>) =>
+    dispatch(setStatus(event.target.value as Status));
+
+  const handleQuery = (event: React.ChangeEvent<HTMLInputElement>) =>
+    dispatch(setQuery(event.target.value));
+
   return (
     <form
       className="field has-addons"
       onSubmit={event => event.preventDefault()}
     >
       <p className="control">
         <span className="select">
-          <select data-cy="statusSelect">
+          <select data-cy="statusSelect" onChange={handleStatus}>
```

Comment:
The value of the options in the select element should match the Status type values. Ensure that the values 'all', 'active', and 'completed' are valid members of the Status enum.